src/main/java/billgates/usecases/user_join/UserJoinUseCase.java

```diff
@@ -0,0 +1,35 @@
+package billgates.usecases.user_join;
+
+import billgates.database.QueryUserData;
+import billgates.interface_adapters.DatabaseGateway;
+
+import java.util.List;
+
+public class UserJoinUseCase implements UserJoinInputPort {
+    private final DatabaseGateway gateway;
+    private final UserJoinOutputPort outputPort;
+
+    public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort) {
+        this.gateway = gateway;
+        this.outputPort = outputPort;
+    }
+
+    @Override
+    public void join(UserJoinRequestModel model) {
+        QueryUserData userData = gateway.getUserData();
+        if (!userData.getUsers().contains(model.getUsername())) {
+            // TODO add the user in the database.
+
+            // TODO log the user in
+        }
+        else {
+            if (userData.getPasswords().contains(model.getPassword())) {
```

Comment:
This is problematic: you should not use `contains` to compare the passwords. For example, assume there is a user called `Scott` with the password `12345678`, and there is another user called `Charlotte` with the password `87654321`. What will happen if the user types in `Scott` in the `UsernameTextField`, but inputs `87654321` in the `passwordTextField`?
According to your code, you will login the user `Scott` even if the user types in the incorrect password.
Since we store the username and the corresponding password at the same index of the lists in `userData`, you could
1. Find the index of this username first
2. Then, use this index to find the corresponding password
3. Finally, compare these two passwords